rfcs/2021-11-22-design-tokens.md

```diff
@@ -0,0 +1,106 @@
+
+A design token solution is needed in which:
+
+1. The set of design tokens remain finite and equal
+2. The set of design tokens maintain consistency between application (DSM, Sketch, React and other future platforms such as iOS, Android, Web Components)
+3. The set of design tokens can be transposed, updated and extended easily and quickly across all systems at once
+4. The set of design tokens can be versioned and changes in that system can be tracked and monitored
+
+By deciding on a [Single Source Of Truth](https://en.wikipedia.org/wiki/Single_source_of_truth) (SSOT) for our design rules, we can organize and transform them for any system which utilizes them.
+
+## Benefits
+
+1. **Consistency** - Between design and code, utilizing a single set of variables everywhere allows a design proof to accurately describe the visual representation of a component as a set of variables. This provides better parity between what's designed and what is seen in the rendered usage on all platforms.
+
+   Looking at the [CSSStats for healthcare.gov/see-plans](https://cssstats.com/stats?url=healthcare.gov/see-plans), there are currently 29 color variations, many of which are nearly identical. Utilizing design tokens could bring this number down considerably and provide better brand consistency.
+
+2. **Organization** - By starting from the atomic level, we can use naming conventions for our tokens that are simple and specific to the variable they represent. By keeping these tokens separate from the systems they are utilized in, they can be managed as a discrete unit. Agreement on naming conventions of variables could eventually create parity between Sketch libraries, DS Sass variables, and future design and development tooling.
+3. **Interoperability** - Design tokens allow a huge amount of flexibility for both moving a design system to a new platform, or creating a new design system. Should an iOS/Android app ever be built, a new web application created, etc. That system would have a framework ready to utilize for its theming which would be consistent with the other brand applications.
+4. **Opportunity for theming** - Decoupling design rules from the codebase allowing the transposition-ing of these static design decisions in various ways, i.e. theming/high contrast. By replacing all tokens in a system with new tokens you can essentially theme swap, assuming the consuming system is utilizing tokens at all levels appropriately.
+
+## Risks
+
+1. There can be a reductive quality to design tokens in that there might be a more limited set of colors/spacing options/etc.. to work with for the consuming applications. Someone might have to sell consuming systems on the advantages to brand consistency, which they hopefully already understand.
+
+## Scope/Outline
+
+**An outline for how this process could proceed:**
+
+1. The SSOT should be decided on, at the moment I see two options:
```

Comment:
Is this an internal decision by the team? Or does it involve outside folks?

Reply:
This goes back to what we started to talk about tangentially yesterday, it would be great if we could do this without the DSM, I offer it as an option because it exists and someone might force the decision? Unsure.

Reply:
I'd expect the core team has a proposal and we get that validated by or CMS partners